#! /usr/bin/env python


from dynamic_reconfigure.parameter_generator_catkin import *


gen = ParameterGenerator()

gen.add("pan_id", int_t, 0, "Motor ID for pan movement", 3, 0, 253)
gen.add("pan_torque", bool_t, 0, "If set, will enable torque", False)
gen.add("pan_pid_p", int_t, 0, "Pan PID p value", 0, 0, 254)
gen.add("pan_pid_d", int_t, 0, "Pan PID d value", 0, 0, 254)
gen.add("pan_pid_i", int_t, 0, "Pan PID i value", 0, 0, 254)
gen.add("pan_position", int_t, 0, "Pan position", 770, 555, 985)
gen.add("pan_playtime", int_t, 0, "Pan playtime", 200, 100, 254)

gen.add("tilt_id", int_t, 0, "Motor ID for tilt movement", 4, 0, 253)
gen.add("tilt_torque", bool_t, 0, "If set, will enable torque", False)
gen.add("tilt_pid_p", int_t, 0, "Tilt PID p value", 0, 0, 254)
gen.add("tilt_pid_d", int_t, 0, "Tilt PID d value", 0, 0, 254)
gen.add("tilt_pid_i", int_t, 0, "Tilt PID i value", 0, 0, 254)
gen.add("tilt_position", int_t, 0, "Tilt position", 845, 800, 890)
gen.add("tilt_playtime", int_t, 0, "Tilt playtime", 200, 100, 254)

exit(gen.generate("elmo", "pan_tilt_config", "PanTilt"))